### PR TITLE
Extract accumulation of received data into separate class

### PR DESCRIPTION
**Pull request by dennisschagt**
Cherry-picked from discarded PR https://github.com/newsboat/newsboat/pull/2306.
This PR will probably cause merge conflicts for https://github.com/newsboat/newsboat/pull/2340. I plan to rebase either of the PRs if the other is merged.

### CODE REVIEW

**Comment by dennisschagt** [comment]
Rebased onto master to make sure all checks will run

**Review by Minoru — APPROVED** [review]
I decided to review this before #2340 to make that other PR smaller and thus easier to review.

The idea is great; thanks for the PR!

Feel free to merge after addressing the merge conflicts and the one small comment I had.

File: mk/newsboat.deps [hunk 1]
```diff
@@ -4,6 +4,7 @@ src/cliargsparser.cpp
 src/configactionhandler.cpp
 src/configpaths.cpp
 src/controller.cpp
+src/curldatareceiver.cpp
```
[the remark this answers is not in the document]
Comment (dennisschagt) [reply mk/newsboat.deps]: Good point, done

**Comment by dennisschagt** [comment]
Thanks! I will enable auto-merge

**Pull request merged**